vita3k/ngs/src/modules/player.cpp

```diff
@@ -172,33 +182,49 @@ bool Module::process(KernelState &kern, const MemState &mem, const SceUID thread
                 decoder->source_frequency = params->playback_frequency;
                 // Enable ADPCM mode on the decoder if needed, and restore state
                 decoder->he_adpcm = static_cast<bool>(params->type);
-                if (decoder->he_adpcm)
-                    std::copy_n(state->adpcm_history, 4, decoder->adpcm_history);
+                if (decoder->he_adpcm) {
+                    if (!decoder->adpcm_history) {
+                        decoder->adpcm_history = new ADPCMHistory[decoder->source_channels]; 
+
+                        ADPCMHistory hist = {};
+                        std::fill_n(decoder->adpcm_history, decoder->source_channels, hist);
+                    }
```

Comment:
Put this part in pcm.cpp instead, the PCM decoder can also be used without ngs (see modules/SceAudiodec/SceAudiodecUser.cpp).